### PR TITLE
HDDS-6171. Create an API to change Bucket Owner

### PR DESCRIPTION
**Pull request by aswinshakil**
## What changes were proposed in this pull request?

Right now, the bucket owner is set when the bucket is created. We need an API to change the owner of a bucket after the bucket is created.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6171

## How was this patch tested?

This patch was tested manually and using integration tests.

### CODE REVIEW

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketArgs.java [hunk 1]
```diff
@@ -230,6 +250,9 @@ public BucketArgs getProtobuf() {
     if(quotaInNamespace > 0 || quotaInNamespace == OzoneConsts.QUOTA_RESET) {
       builder.setQuotaInNamespace(quotaInNamespace);
     }
+    if(ownerName != null) {
```
Comment (smengcl) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketArgs.java:253]:
```suggestion
    if (ownerName != null) {
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClientAbstract.java [hunk 2]
```diff
@@ -297,6 +297,26 @@ public void testVolumeSetOwner() throws IOException {
...
+    String ownerName = "testUser";
+
+    ClientProtocol proxy = store.getClientProxy();
+    proxy.setBucketOwner(volumeName, bucketName, ownerName);
+    String newOwner = store.getVolume(volumeName).getBucket(bucketName)
+        .getOwner();
+
+    assertEquals(ownerName, newOwner);
+    assertNotEquals(oldOwner, newOwner);
+  }
```
Comment (smengcl) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClientAbstract.java:318]: Would you add a clean up logic to this test case? i.e. delete the bucket and volume created in this test case at the end.

**Review by smengcl — COMMENTED** [review]
Thanks for the patch @aswinshakil

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/protocol/ClientProtocol.java [hunk 3]
```diff
@@ -772,4 +772,14 @@ OzoneKey headObject(String volumeName, String bucketName,
...
   void clearTheadLocalS3Auth();
+
+  /**
+   * Sets the owner of bucket.
+   * @param volumeName Name of the Volume
+   * @param bucketName Name of the Bucket
+   * @param owner to be set for the bucket
+   * @throws IOException
+   */
+  void setBucketOwner(String volumeName, String bucketName,
```
[the remark this answers is not in the document]
Comment (aswinshakil) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/protocol/ClientProtocol.java:783]: Yes I can add a response for `SetBucketPropertyResponse` to return boolean in `OmClientProtocol.proto`

**Review by smengcl — APPROVED** [review]
+1 pending CI

**Pull request merged**